binstar_client/commands/upload.py

```diff
@@ -73,23 +69,23 @@ def determine_package_type(filename, args):
     -t/--package-type argument
     """
     if args.package_type:
-        package_type = args.package_type
-    else:
-        logger.info('Detecting file type...')
+        return PackageType(args.package_type)
 
-        package_type = detect_package_type(filename)
+    logger.info('Detecting file type...')
+    package_type = detect_package_type(filename)
 
-        if package_type is None:
-            message = 'Could not detect package type of file %r please specify package type with option --package-type' % filename
-            logger.error(message)
-            raise errors.BinstarError(message)
+    if package_type is None:
+        message = 'Could not detect package type of file %r please ' \
+                  'specify package type with option --package-type' % filename
+        logger.error(message)
+        raise errors.BinstarError(message)
 
-        logger.info('File type is "%s"', package_type)
+    logger.info('File type is "{}"'.format(package_type.label()))
```

Comment:
It's better to allow logger perform formatting on its own (like it was done before changes).